lightning/src/ln/channelmanager.rs

```diff
@@ -1334,6 +1340,57 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 		})
 	}
 
+	#[allow(dead_code)]
+	// Messages of up to 64KB should never end up more than half full with addresses, as that would
+	// be absurd. We ensure this by checking that at least 500 (our stated public contract on when
```

Comment:
Hm, this isn't an actual spec rule, though, is it (mod the issue that [this PR](https://github.com/lightningnetwork/lightning-rfc/pull/751) addresses)? >500 does seem extreme and I don't have an issue with enforcing it, just not sure the exact purpose of enforcing it...

Reply:
IIRC (and if it doesn't it should) if we went to serialize it we'd generate something >64KB, panicing trying to send a message that overflows the max message size.